solidity/contracts/bridge/Bridge.sol

```diff
@@ -144,6 +165,14 @@ contract Bridge is Ownable {
     ///         prefix refers to the transaction which created that main UTXO.
     mapping(bytes20 => bytes32) public mainUtxos;
 
+    // TODO: Documentation.
+    // Key: keccak256(walletPubKeyHash | redeemerOutputHash)
+    mapping(uint256 => RedemptionRequest) public redemptionRequests;
+
+    // TODO: Documentation.
+    // Key: keccak256(walletPubKeyHash | redeemerOutputHash)
+    mapping(uint256 => bool) public redemptionFaults;
```

Comment:
This is not only about requests. If an arbitrary key (wallet pubkey hash + output hash) is proven to be a non-requested redemption (thus fraud) it is also pushed here.

Reply:
What if we stored non-requested redemption frauds and timed-out redemption frauds separately? I think it could let us reuse the redemption wallet in case there was already a fraud related to it?

For non-requested redemption fraud, we'd probably need to add the transaction hash to the identifier.
For timed-out redemption fraud, we'd probably need to add the time the redemption was requested at to the identifier.

Another advantage - but I am still reviewing the code so not sure about it - would be to let to report non-requested redemption frauds for the given BTC address several times if it happened several times (in separate transactions).

Reply:
As we discussed out-of-band, we'll keep the mapping for "pending requests" and "faulty redemptions" as presented in this pull request. In case a wallet commit a fault (accidentally or not), we block new redemption requests targeting that wallet and we expect the wallet (if possible) to transfer their outstanding funds to another healthy wallet.